mm2src/mm2_bitcoin/script/src/script.rs

```diff
@@ -420,6 +420,7 @@ impl Script {
                     _ => unreachable!(), // because we are relying on script_type() checks here
                 })
                 .map(|public| {
+                    // DISCUSS: Should we really reduce this to pkh? what is it used for?
```

Comment:
If we will merge this note, it would be great if we can change this to `TODO`. I often grep this placeholder in the entire codebase to see what we left behind.
```suggestion
                    // TODO: Discussion needed: "Should we really reduce this to pkh? what is it used for?"
```

Reply:
> Should we really reduce this to pkh?
btw you can check the original ExtractDestination in the daemon code, I can see it works in the similar way as basically used to get the address from P2PK utxos, to add the utxo to the wallet balance

Reply:
> btw you can check the original ExtractDestination in the daemon code, I can see it works in the similar way as basically used to get the address from P2PK utxos, to add the utxo to the wallet balance

iirc, yeah, it's used somewhere to extract the address the coins are sent to, but they are just used for comparison and not stored anymore (basically, is this output sent our address?). the optional `pubkey` part doesn't contribute to that comparison ([see](https://github.com/mariocynicys/adex/blob/f07e720ad8d63c189d31200054f8e382b862aa98/mm2src/mm2_bitcoin/keys/src/address.rs#L120-L144)).

maybe we can check if this output belongs to us by comparing output scripts instead?